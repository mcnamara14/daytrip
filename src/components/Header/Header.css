header {
  width: 1200px;
  display: flex;
  justify-content: space-between;
  margin: 0 auto;
  height: 75px;
  align-items: center;
}

header img {
  height: 46px;
}

nav {
  width: 40%;
  display: flex;
  justify-content: space-between;
}

nav a {
  font-family: "Exo 2";
  font-size: 1.1rem;
  text-transform: uppercase;
  text-decoration: none;
  font-weight: 500;
  color: #98b5c8;
  font-style: italic;
}

.loginLink {
  display: flex;
  justify-content: space-between;
  width: 20%;
  align-items: center;
  margin-left: 55px;
}

.loginLink img {
  height: 25px;
}