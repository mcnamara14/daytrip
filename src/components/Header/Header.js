import React from 'react';
import './Header.css';

export const Header = () =>  {
  return (
    <header>  
      I'm a header
    </header>
  );
}
export default Header;
