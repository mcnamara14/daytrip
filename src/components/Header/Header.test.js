import React from 'react';
import ReactDOM from 'react-dom';
import { Header, mapDispatchToProps } from './Header';
import { shallow } from 'enzyme';

describe('Header', () => {
  it('should match the snapshot', () => {
    const mockUser = {
      userId: '12345'
    }

    const wrapper = shallow(<Header user={mockUser} />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('signoutLoginClickHandler', () => {
    it('should call loginUser with correct args', () => {
      const mockUser = {
        userId: '12345'
      }
    
      const wrapper = shallow(<Header user={mockUser} loginUser={jest.fn()} />)
      const expected = [{"city": "", "email": "", "state": "", "userId": null}]
      wrapper.instance().signoutLoginClickHandler();

      expect(wrapper.instance().props.loginUser).toHaveBeenCalledWith(...expected);
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the correct params on loginUser', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'LOGIN_USER',
        userId: '12345',
        email: 'test@testerson.com',
        city: 'boulder',
        state: 'co'
      };

      mappedProps.loginUser('12345', 'test@testerson.com', 'boulder', 'co');

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });
  });
})

