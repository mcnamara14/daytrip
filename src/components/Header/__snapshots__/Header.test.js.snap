// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Header logo"
          className="headerLogo"
          src="header-logo.jpg"
/>,
        <nav>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/events"
          >
                    Events
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/trips"
          >
                    Trips
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/account"
          >
                    Account
          </NavLink>
          <div
                    className="loginLink"
          >
                    <img
                              alt="Login icon"
                              className="loginIcon"
                              src="header-login-icon.jpg"
                    />
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="navLogin"
                              to="/login"
                    >
                              Login
                    </NavLink>
          </div>
</nav>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Header logo",
          "className": "headerLogo",
          "src": "header-logo.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/events"
>
              Events
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/trips"
>
              Trips
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/account"
>
              Account
</NavLink>,
            <div
              className="loginLink"
>
              <img
                            alt="Login icon"
                            className="loginIcon"
                            src="header-login-icon.jpg"
              />
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="navLogin"
                            to="/login"
              >
                            Login
              </NavLink>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Events",
              "to": "/events",
            },
            "ref": null,
            "rendered": "Events",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Trips",
              "to": "/trips",
            },
            "ref": null,
            "rendered": "Trips",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Account",
              "to": "/account",
            },
            "ref": null,
            "rendered": "Account",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="Login icon"
                  className="loginIcon"
                  src="header-login-icon.jpg"
/>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="navLogin"
                  to="/login"
>
                  Login
</NavLink>,
              ],
              "className": "loginLink",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Login icon",
                  "className": "loginIcon",
                  "src": "header-login-icon.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Login",
                  "className": "navLogin",
                  "to": "/login",
                },
                "ref": null,
                "rendered": "Login",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "nav",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Header logo"
            className="headerLogo"
            src="header-logo.jpg"
/>,
          <nav>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/events"
            >
                        Events
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/trips"
            >
                        Trips
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/account"
            >
                        Account
            </NavLink>
            <div
                        className="loginLink"
            >
                        <img
                                    alt="Login icon"
                                    className="loginIcon"
                                    src="header-login-icon.jpg"
                        />
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="navLogin"
                                    to="/login"
                        >
                                    Login
                        </NavLink>
            </div>
</nav>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Header logo",
            "className": "headerLogo",
            "src": "header-logo.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/events"
>
                Events
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/trips"
>
                Trips
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/account"
>
                Account
</NavLink>,
              <div
                className="loginLink"
>
                <img
                                alt="Login icon"
                                className="loginIcon"
                                src="header-login-icon.jpg"
                />
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                className="navLogin"
                                to="/login"
                >
                                Login
                </NavLink>
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Events",
                "to": "/events",
              },
              "ref": null,
              "rendered": "Events",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Trips",
                "to": "/trips",
              },
              "ref": null,
              "rendered": "Trips",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Account",
                "to": "/account",
              },
              "ref": null,
              "rendered": "Account",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="Login icon"
                    className="loginIcon"
                    src="header-login-icon.jpg"
/>,
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="navLogin"
                    to="/login"
>
                    Login
</NavLink>,
                ],
                "className": "loginLink",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Login icon",
                    "className": "loginIcon",
                    "src": "header-login-icon.jpg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "Login",
                    "className": "navLogin",
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "nav",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
